src/main/cluster-detectors/detector-registry.ts

```diff
@@ -0,0 +1,46 @@
+import { observable } from "mobx";
+import { ClusterMetadata } from "../../common/cluster-store";
+import { Cluster } from "../cluster";
+import { BaseClusterDetector, ClusterDetectionResult } from "./base-cluster-detector";
+import { ClusterIdDetector } from "./cluster-id-detector";
+import { DistributionDetector } from "./distribution-detector";
+import { LastSeenDetector } from "./last-seen-detector";
+import { NodesCountDetector } from "./nodes-count-detector";
+import { VersionDetector } from "./version-detector";
+
+export class DetectorRegistry {
+  registry = observable.array<typeof BaseClusterDetector>([], { deep: false });
+
+  add(detectorClass: typeof BaseClusterDetector) {
+    this.registry.push(detectorClass)
+  }
+
+  async detectForCluster(cluster: Cluster): Promise<ClusterMetadata> {
+    const results: {[key: string]: ClusterDetectionResult } = {}
+    for (const detectorClass of this.registry) {
+      const detector = new detectorClass(cluster)
+      try {
+        const data = await detector.detect()
+        if (!data) continue;
+        const existingValue = results[detector.key]
+        if ((existingValue && existingValue.accuracy < data.accuracy) || !existingValue) { // previous value exists and is less accurate
+          results[detector.key] = data
+        }
```

Comment:
Is this cleaner?
```suggestion
        if (existingValue && existingValue.accuracy > data.accuracy) continue; // previous value exists and is more accurate
        results[detector.key] = data
```
I think this also ensures that the "freshest" value is kept in case the accuracy is the same, right?